api/src/stampy_chat/chat.py

```diff
@@ -340,9 +344,38 @@ def run_query(session_id: str, query: str, history: List[Dict], settings: Settin
         prompt=make_prompt(settings, chat_model, callbacks),
         memory=make_memory(settings, history, callbacks)
     )
+
     if followups:
         chain = chain | StampyChain(callbacks=callbacks)
+
     result = chain.invoke({"query": query, 'history': history}, {'callbacks': []})
+
+    #Validate results
+    contexts=[c['text'] for c in get_top_k_blocks(query, 5)]
+    rag_response={
+            "llm_answer": result['text'],
+            "llm_context_list": contexts
+            }
+
+    benchmark = BenchmarkItem(question=query)
+
+    llm_response = LLMResponse(
+            llm_answer=result['text'],
+            llm_context_list=contexts,
+            benchmark_item=benchmark
+        )
+
+    # Score the responses
+    scorer = ValidateScorer([
+        AnswerConsistencyMetric(),
+        AugmentationAccuracyMetric(),
+        RetrievalPrecisionMetric()
+    ])
+    run = scorer.score_responses([llm_response])
```

Comment:
is this actually returned to the user? If not, then I'd suggest doing it after notifying the frontend that everything is done, i.e. moving the `if callback (...)` clause before these

Reply:
Makes sense. I will make this change. The reason I kept it this way is because I was worried that a "fast" user might ask the next question before the evaluation is done running. This might overwhelm the system and spawn a lot of processes.

I don't think this will be useful for our users. We may want to track it internally first and then decide what would be the best way to display this to users. For instance, I don't think users will know what "context-precision" is or care about it.